### PR TITLE
Remove unused LLVM related code

### PR DESCRIPTION
**Pull request by dotdash**
Ticks a few more boxes on #46437

### CODE REVIEW

File: src/librustc/session/config.rs [hunk 1]
```diff
@@ -1082,8 +1082,6 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
         "gather borrowck statistics"),
     no_landing_pads: bool = (false, parse_bool, [TRACKED],
         "omit landing pads for unwinding"),
-    debug_llvm: bool = (false, parse_bool, [UNTRACKED],
```
Comment (rkruppe) [inline src/librustc/session/config.rs]: Is the idea behind removing this flag that one can pass `-Cllvm-args=-debug` instead? I'm fine with that, but it's not exactly "unused".

**Review by hanna-kruppe — COMMENTED** [review]
LGTM except removing the `-Z` flag.

File: src/librustc/session/config.rs (hunk 1 above)
Comment (rkruppe) [reply src/librustc/session/config.rs]: Oh yes, now I see that the commit message mentions precisely that. As I said I'm fine with it but it's kind of a visible change so I don't feel comfortable giving r+ myself.

**Comment by alexcrichton** [comment]
@bors: r+

Nice!

**Comment by dotdash** [comment]
Looks like a timeout? I don't see any errors and the log seems to be cut off

**Comment by alexcrichton** [comment]
@bors: retry

* https://github.com/rust-lang/rust/issues/46903

**Pull request merged**